Windows - guidance for shell mode in the REPL

reumle opened the pull request:
Shell mode behavior is currently very non intuitive on Windows. It essentially does nothing a basic user (that would be me..) understands by shell mode (list directory, show environment variables etc) . So I clarified, and showed a way to fire up the two most common shells, `cmd.exe` and `powershell`.

reumle committed:
Windows - guidance for shell mode in the REPL

The REPL shell mode is very non intuititive for basic Windows users: the mode does nothing  (like list environment variables, list directory) such a user expects, and throws errors as well. So I clarify the behavior, and show how to access one such Windows shell that most users understand under "shell mode".

StefanKarpinski commented on stdlib/REPL/docs/src/index.md:
````diff
@@ -135,7 +135,23 @@ julia> ; # upon typing ;, the prompt changes (in place) to: shell>
 shell> echo hello
 hello
 ```
+* Note for Windows users: Julia Shell mode does not expose windows shell commands.Hence, this will fail: 
````
```suggestion
!!! note
    For Windows users, Julia's shell mode does not expose windows shell commands.
    Hence, this will fail:
```

reumle replied:
closing (?)  this pull request, in order to resubmit with your suggestions, and other small enhancements (failing test etc), in a proper branch this time (submission guidelines). Thanks for reviewing!

StefanKarpinski replied:
You can push new commits to the existing pull request. It's much easier if you just commit the suggestions on the same branch.

reumle committed:
Update index.md

reumle commented:
This now passes (10/12) tests (2 still running), and shows how to start `cmd.exe`, the second, legacy, windows shell. 

Small caveat, both shells are in french… I will try to get English texts… but it will need another commit..

reumle committed:
Windows - guidance for shell mode in the REPL

As mentionned in my last comment: shell examples are now English speaking.
No other change.

reumle commented:
Not sure what to do about the 2 failing tests...

ViralBShah commented:
Thanks @reumle. I ran into these issues on Windows myself, so it is good to have these.
